.github/workflows/test.yml

```diff
@@ -55,9 +56,35 @@ jobs:
           POSTGRES_HOSTNAME: psql-a
           POSTGRES_HOSTNAME_A: psql-a
           POSTGRES_HOSTNAME_B: psql-b
+          
+  # Run unit tests on Linux Swift runners on 
+  linux-unit-tests:
+    strategy:
+      fail-fast: false
+      matrix:
+        swiftver:
+          - swift:5.2
+          - swift:5.3
+          - swift:5.4
+          - swiftlang/swift:nightly-5.5
+          - swiftlang/swift:nightly-main
+        swiftos:
+          #- xenial
+          #- bionic
+          - focal
+          #- centos7
+          #- centos8
+          #- amazonlinux2
```

Comment:
I think for the unit tests we should be testing against _all_ supported OS versions rather than just one.

Reply:
The reason I disabled the other OS' was that there are several workarounds necessary for, e.g., SPM support on CentOS 7, SQLite support on AmazonLinux2, etc., plus the full test matrix with all versions _and_ OS' yields well over 100 tests _per push_ - it was just too many, and it was causing stalls for CI in other repos since the "workflows in progress at once" limit is per-org, not per-repo.

Reply:
(Yes, I realize the matrix given here would only yield 30 tests, but then also multiply that by the multiple postgresql versions - with 11, 12, and 13, that was already 90 tests right there. And then also two more (latest and latest-stable) per PSQL version for macOS gives 100, and at the time I was also trying to get both Catalina and Big Sur macOS tests going... it starts really adding up.)

Reply:
Cool fine by me